.github/workflows/ci.yml

```diff
@@ -13,36 +14,80 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - uses: the-guild-org/shared-config/setup@main
         name: Setup Env
         with:
           nodeVersion: 20
           packageManager: pnpm
-          packageManagerVersion: 9.1.0
+          packageManagerVersion: 9.5.0
 
       - name: Lint
         run: pnpm lint
 
+  build-matchstick:
+    name: Build Matchstick
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.77.0
+
+      - uses: the-guild-org/shared-config/setup@main
+        name: Setup Env
+        with:
+          nodeVersion: 20
+          packageManager: pnpm
+          packageManagerVersion: 9.5.0
+
+      - name: Install Dependencies
+        run: pnpm install
+
+      - name: Install matchstick system dependencies
+        working-directory: packages/matchstick
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install libpq-dev protobuf-compiler
+
+      - name: Build Matchstick Package
+        run: pnpm --filter=@graphprotocol/graph-tooling-napi-utils build
+
+      - name: Upload Matchstick Build Artifacts
+        uses: actions/upload-artifact@v3
+        with:
```

Comment:
does GitHub automatically add commit refs to these so in the next step u are pulling right artifact?

Reply:
based on previous runs, yeah, it seems to be pulling the artifacts correctly
https://github.com/graphprotocol/graph-tooling/actions/runs/9949158544/job/27485115091